### PR TITLE
feat: org owner-admin cant see the teams created by other owner-admin 9982 cal 2120 

### LINKED ISSUE #9982: [CAL-2120] Org owner/admin can't see the teams created by other owner/admin
He should be able to see all teams so that he can add himself to any team he wants.
He shouldn't be able to see a team's event-types unless he has added him to the team.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2120](https://linear.app/calcom/issue/CAL-2120/org-owneradmin-cant-see-the-teams-created-by-other-owneradmin)</sub>

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- If user is owner/admin in org in settings tabs it should displays teams that other owner/admin created but current user don't belong to.

Fixes #9982 

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How should this be tested?

- Are there environment variables that should be set? No
- What are the minimal test data to have? Org teams with two owner/admins
- What is expected (happy path) to have (input and output)? One of the two owner/admins should create a new team excluding the other one and from the excluded user he should be able to see other teams created in org.
- Any other important info that could help to test that PR

## Mandatory Tasks

- [X] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/ee/teams/components/MemberListItem.tsx [hunk 1]
```diff
@@ -97,11 +97,11 @@ export default function MemberListItem(props: Props) {
     });
```
Comment (alannnc) [inline packages/features/ee/teams/components/MemberListItem.tsx:97]: Undo changes on this file.

File: packages/features/ee/teams/components/TeamInviteListItem.tsx [hunk 2]
```diff
@@ -58,32 +58,15 @@ export default function TeamInviteListItem(props: Props) {
...
-      <div className="ms-3 inline-block">
-        <span className="text-emphasis text-sm font-semibold">{team.name}</span>
-        <span className="text-default block text-sm leading-5">
-          {t("invited_by_team", { teamName: team.name, role: t(team.role.toLocaleLowerCase()) })}
-        </span>
-      </div>
-    </div>
-  );
-
+  console.log("team invite list", team);
```
Comment (alannnc) [inline packages/features/ee/teams/components/TeamInviteListItem.tsx:61]: Undo changes on this file.

File: packages/features/ee/teams/pages/team-listing-view.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { Meta } from "@calcom/ui";
 import { getLayout } from "../../../settings/layouts/SettingsLayout";
 import { TeamsListing } from "../components";
 
-const BillingView = () => {
+const TeamListingView = () => {
```
Comment (alannnc) [inline packages/features/ee/teams/pages/team-listing-view.tsx:7]: Getting the right name for the component.

File: packages/features/ee/organizations/pages/settings/other-team-members-view.tsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+const MembersView = () => {
+  const { t, i18n } = useLocale();
+  const router = useRouter();
+
+  const teamId = Number(router.query.id);
+
+  const utils = trpc.useContext();
+  const [offset, setOffset] = useState<number>(1);
+  // const [query, setQuery] = useState<string | undefined>("");
```
Comment (alannnc) [inline packages/features/ee/organizations/pages/settings/other-team-members-view.tsx:57]: There will some code commented on this file until I make it work.

File: packages/features/ee/organizations/pages/settings/other-team-members-view.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+  // const [queryToFetch, setQueryToFetch] = useState<string | undefined>("");
+  const [loadMore, setLoadMore] = useState<boolean>(true);
+  const limit = 100;
```
Comment (alannnc) [inline packages/features/ee/organizations/pages/settings/other-team-members-view.tsx:60]: Limiting temporarily, till paginate works.

File: packages/trpc/server/routers/viewer/teams/update.handler.ts [hunk 6]
```diff
@@ -19,7 +19,13 @@ type UpdateOptions = {
 };
 
 export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
-  if (!(await isTeamAdmin(ctx.user?.id, input.id))) throw new TRPCError({ code: "UNAUTHORIZED" });
+  const isOrgAdmin = ctx.user?.organization?.isOrgAdmin;
+
+  if (!isOrgAdmin) {
+    if (!(await isTeamAdmin(ctx.user?.id, input.id))) {
+      throw new TRPCError({ code: "UNAUTHORIZED" });
```
Comment (alannnc) [inline packages/trpc/server/routers/viewer/teams/update.handler.ts:26]: First we validate if !orgAdmin then we validate if !isTeamAdmin.

**Review by alannnc — COMMENTED** [review]
Created new components/trpc calls since I didn't want to reuse teams logic since future updates on those could affect our new ones. It's also a good thing to keep code on it's own domain.

- Partially hide member search function and team appearance edit.

**Review by sean-brydon — DISMISSED** [review]
Hey - i know this is in draft but i don't see why we can't re-use the pages/components we currently use for teams. They are gonna be displaying the exact same data.

**Comment by alannnc** [comment]
> Hey - i know this is in draft but i don't see why we can't re-use the pages/components we currently use for teams. They are gonna be displaying the exact same data.
@sean-brydon 

Tried but didn't like the solution since I felt I was juggling on the server to get the data for 2 scenarios, this means components there are highly tied up to teams trpcs data. Domain wise orgs feature shouldn't import directly from teams, this can lead to bad things.
Probably moving some components from there to a generic and then reusing them.

File: packages/features/ee/organizations/pages/components/OtherTeamsListing.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import { useRouter } from "next/router";
```
Comment (zomars) [inline packages/features/ee/organizations/pages/components/OtherTeamsListing.tsx:1]: Same here

**Review by zomars — DISMISSED** [review]

File: packages/features/ee/organizations/pages/settings/other-team-members-view.tsx [hunk 8]
```diff
@@ -0,0 +1,214 @@
+// import { debounce } from "lodash";
+import { useRouter } from "next/router";
```
[the remark this answers is not in the document]
Comment (alannnc) [reply packages/features/ee/organizations/pages/settings/other-team-members-view.tsx:2]: I'm getting teamId from the url param, seems like useSearchParams only works for queries and not from paths?
https://nextjs.org/docs/app/api-reference/functions/use-search-params
@zomars

[the remark this answers is not in the document]
Comment (alannnc) [reply packages/features/ee/organizations/pages/settings/other-team-members-view.tsx:2]: nvm figured it out

**Review by alannnc — COMMENTED** [review]
@zomars Changes for useRouter are done.

**Review by leog — DISMISSED** [review]
Code LGTM 👍

**Review by PeerRich — APPROVED** [review]
reapproving on behalf of leo's past review

**Pull request merged**